libs/lb/lb-rs/src/rpc.rs

```diff
@@ -0,0 +1,139 @@
+#[derive(Serialize, Deserialize)]
+pub struct RpcRequest {
+    pub method: String,
+    pub args: Option<Vec<u8>>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum CallbackMessage<S, T> {
+    Status(S),
+    Done(LbResult<T>),
+}
+
+impl RpcRequest {
+    pub fn new(method: impl Into<String>, args: Option<Vec<u8>>) -> Self {
+        Self {
+            method: method.into(),
+            args,
+        }
+    }
+}
+
+pub async fn call_rpc<T>(
+    socket_address: SocketAddrV4,
+    method: &str,
+    args: Option<Vec<u8>>,
+) -> LbResult<T>
+where
+    T: for<'de> Deserialize<'de>,
+{
+    let mut stream = TcpStream::connect(socket_address)
+            .await
+            .map_err(core_err_unexpected)?;
+    let req = RpcRequest::new(method, args);
+    let encoded = bincode::serialize(&req).map_err(core_err_unexpected)?;
+    let len = encoded.len() as u32;
```

Comment:
This limits the maximum request size to about 4gb, which to be fair many parts of our platform would struggle with at the moment. But I think you should not impose an additional limitation here and shoot for the full size of a `usize`.